client/src/components/subscription/subscriptionForm.js

```diff
@@ -0,0 +1,241 @@
+import React from 'react';
+import { toast } from 'react-toastify';
+import ReactDayPicker from '../date/datepicker';
+import convertWeekdaysToDates from '../../utils/frontendDateUtils';
+
+function convertStringToDate(datesArray) {
+  return datesArray.map((date) => new Date(date));
+};
```

Comment:
can one-liner this fxn using a `const`

```suggestion
const convertStringToDate = (datesArray) => datesArray.map((date) => new Date(date));
```